### PR TITLE
Add Type Definitions File

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
This precedes: https://github.com/exivity/react-orbitjs/pull/16

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
currently applying feedback from other PR

**Comment by NullVoxPopuli** [comment]
@hongaar ready for review

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  dataStore: Store;
+}
+
+export interface RecordsToProps {
+  [key: string]: (q: QueryBuilder) => any;
+}
+
+export interface WithData {
+  dataStore: Store;
+  sources: { [sourceName: string]: Source };
```
Comment (hongaar) [inline index.d.ts]: If you remove this line it's ready to merge!

**Comment by NullVoxPopuli** [comment]
updated.
I'm sure I'll need to submit additional PRs tweaking these types over time as well. :)

**Comment by hongaar** [comment]
Thanks @NullVoxPopuli. Available as [v0.1.10](https://www.npmjs.com/package/react-orbitjs/v/0.1.10).

**Comment by NullVoxPopuli** [comment]
Cool, now, we can talk about the sources :)

**Pull request merged**